### DISCUSSION TITLE
Copilot tends to use only one library for each sentence.

### DISCUSSION THREAD

**Discussion by GWMCwing** [community/community#7165] [Copilot Conversations]
When I am using python with Copilot, it tends to not using the mentioned built-in libraries, and keeps extending the sentence, or leaving the suggestions with the libraries at the end of the suggestion list, if the libraries is not imported before hand. 

E.g. Copilot won't import random, when commenting "moving turtle with random amount of steps". It seems like when having more than one libraries are required in a sentence, Copilot tends to use only the first recognized library (Turtle), while leaving the second one (random) unused.

**Comment by figloalds** [comment]
Copilot has a harder time crafting suggestions using little known libraries
You can try to "tell" it to use a certain library commenting like
"using <such> library"
for example, if I write in javascript
const fetchAddressFromPostalCode = ...
It will suggest some api call with fetch or XHR, i can import axios and then comment "fetch address data using axios" to hint it to use axios and then it will do.

Again, it has a way easier time suggesting widely used libraries

**Reply by GWMCwing** [reply]
I do understand copilot is not used for using unknown libraries functions.
But for built in functions and libraries in python, it either used the library's function without importing them, or importing unrelated libraries like numpy for random numbers function or even statistics related functions
